vm/systemSmartContracts/esdt.go

```diff
@@ -251,26 +254,51 @@ func (e *esdt) issueToken(owner []byte, arguments [][]byte) error {
 	return nil
 }
 
-func (e *esdt) upgradeProperties(token *ESDTData, args [][]byte) {
-	for _, arg := range args {
-		optionalArg := string(arg)
+func upgradeProperties(token *ESDTData, args [][]byte) error {
+	if len(args) == 0 {
+		return nil
+	}
+	if len(args)%2 != 0 {
+		return vm.ErrInvalidNumOfArguments
+	}
+
+	for i := 0; i < len(args); i += 2 {
+		optionalArg := string(args[i])
+		val, err := checkAndGetSetting(string(args[i+1]))
+		if err != nil {
+			return err
+		}
 		switch optionalArg {
 		case burnable:
-			token.Burnable = true
+			token.Burnable = val
 		case mintable:
-			token.Mintable = true
+			token.Mintable = val
 		case canPause:
-			token.CanPause = true
+			token.CanPause = val
 		case canFreeze:
-			token.CanFreeze = true
+			token.CanFreeze = val
 		case canWipe:
-			token.CanWipe = true
+			token.CanWipe = val
 		case upgradable:
-			token.Upgradable = true
+			token.Upgradable = val
```

Comment:
This has to be treated cautiously by the the owner of the esdt token. He/she might wrongly set upgradable = false with some unwanted settings that will then become impossible to be changed in the future.

Reply:
this is the same with upgradable contract. by default upgradable is true.